pypdf/_writer.py

```diff
@@ -3105,6 +3106,7 @@ def find_outline_item(
                 o = cast(TreeObject, o["/Next"])
             else:
                 return None
+        return
```

Comment:
There are two issues here:

* We always return `None` - why does this check trigger at all?
* If this is a valid case, we should have a test for it as well.

Reply:
Looking at this again, this might indeed be correct and possibly even tested.